contracts/prize-strategy/multiple-winners/MultipleWinners.sol

```diff
@@ -10,10 +10,24 @@ contract MultipleWinners is PeriodicPrizeStrategy {
 
   bool public splitExternalErc20Awards;
 
+  mapping(address => bool) public isBlocklisted;
+
+  bool public carryOverBlocklist;
+
+  uint256 public blocklistRetryCount;
+
   event SplitExternalErc20AwardsSet(bool splitExternalErc20Awards);
 
   event NumberOfWinnersSet(uint256 numberOfWinners);
 
+  event BlocklistCarrySet(bool carry);
+
+  event BlocklistSet(address indexed user, bool blocklisted);
+
+  event BlocklistRetryCountSet(uint256 count);
+
+  event RetryMaxLimitReached(uint256 numberOfWinners);
```

Comment:
The original event included the `__numberOfWinners` variable in the event. Even though you can easily lookup that information via `numberOfWinners` the number could be stale if the number of winners was changed.

Without an archive node it's not possible to determine the accurate `__numberOfWinners` if the the variable has been changed during the PrizeStrategy lifecycle when the event was emitted.

Maybe it makes sense to include `__numberOfWinners` to avoid requiring an archive node or comparing the timestamp in the `NumberOfWinnersSet` event?

Reply:
The event will be captured in one of two scenarios:

1. They’re listening in real-time.  At which point they’ll be able to query the current state of retries & winners
2. They’re rebuilding the state from history (ala subgraph) at which point they’ll have all of the update events for retries and num winners

So we should keep the code DRY and *not* emit the numberOfWinners.  **However** we should emit the *actual* winner count, since that's an ephemeral value that might be interesting.  See comment below